enumerables.rb

```diff
@@ -0,0 +1,102 @@
+module Enumerable
+  def my_each
+    size = count - 1
+    counter = 0
+    items = to_a
+    while counter <= size
+      if is_a? Range
+        items.length >= 0 && yield(items[counter])
+      elsif is_a? Hash
+        yield(items[counter][0], items[counter][1])
+      else
+        yield(self[counter])
+      end
+      counter += 1
+    end
+    self
+  end
+
+  def my_each_with_index()
+    counter = 0
+    size = length - 1
+    while counter <= size
+      yield(self[counter], counter)
+      counter += 1
+    end
+    self
+  end
+
+  def my_select
+    required = []
+    my_each do |r|
+      required.push(yield(r))
+    end
+    required
+  end
+
+  def my_all?
+    track_negative = 0
+    my_each do |t|
+      yield(t) == true ? true : track_negative += 1
+    end
+    !track_negative.positive?
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true when none of the collection members are false or nil
```ruby
true_array = [1, true, 'hi', []] 
false_array = [1, false, 'hi', []] 
true_array.my_all? == true_array.all? #true
```
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
```ruby
array = Array.new(100){rand(0...9)}
array.my_all?(Integer) == array.all?(Integer)
```
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
```ruby
words = %w[dog door rod blade]
words.my_all?(/d/) == words.all?(/d/)
```
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
```ruby
array.my_all?(3) == array.all?(3) #true
```